src/fsharp/NameResolution.fs

```diff
@@ -2331,7 +2334,21 @@ let rec private ResolveTypeLongIdentInModuleOrNamespace (ncenv:NameResolver) (ty
         let tcrefs = LookupTypeNameInEntityMaybeHaveArity (ncenv.amap, id.idRange, id.idText, typeNameResInfo.StaticArgsInfo, modref)
         match tcrefs with 
         | _ :: _ -> tcrefs |> CollectResults (fun tcref -> success(resInfo,tcref))
-        | [] -> raze (UndefinedName(depth,FSComp.SR.undefinedNameType,id,[]))
+        | [] ->
+            let possibleTypes =
```

Comment:
Please move this new code into a new top-level function (e.g. ``SuggestTypeLongIdentInModuleOrNamespace``). In general I'd like to see all "error correction" code to be separated into new top-level functions rather, so that the main body of a function like ``ResolveTypeLongIdentInModuleOrNamespace`` concentrates on the positive logic, and can be compared with the F# language specification text that corresponds to these cases.